commit 2310a56c794e2e6cc93305c7c45e51fa3ed469f0
Author: Andrew Price <anprice@redhat.com>
Date:   Mon Jul 8 10:31:24 2013 +0100

    init.d/gfs2: Work around nested mount points umount bug
    
    Work around a bug where umounting nested gfs2 mount points would take
    place in the wrong order.
    
    Resolves: rhbz#963657
    
    Signed-off-by: Andrew Price <anprice@redhat.com>

diff --git a/gfs2/init.d/gfs2.in b/gfs2/init.d/gfs2.in
index 0f0f009..9199d2c 100644
--- a/gfs2/init.d/gfs2.in
+++ b/gfs2/init.d/gfs2.in
@@ -68,7 +68,7 @@ fi
 # This script's behavior is modeled closely after the netfs script.  
 #
 GFS2FSTAB=$(LC_ALL=C awk '!/^#/ && $3 == "gfs2" && $4 !~ /noauto/ { print $2 }' /etc/fstab)
-GFS2MTAB=$(LC_ALL=C awk '!/^#/ && $3 == "gfs2" && $2 != "/" { print $2 }' /proc/mounts)
+GFS2MTAB=$(LC_ALL=C awk '!/^#/ && $3 == "gfs2" && $2 != "/" { print $2 }' /proc/mounts | sort -r)
 
 if [ -z "$GFS2FSTAB" ]; then
 	echo "GFS2: no entries found in /etc/fstab"
